pyocm/pyocm.py

```diff
@@ -5,41 +5,46 @@
 
 class ocm_api:
     """."""
-    def __init__(self, api="https://api.openshift.com/api/", 
+    def __init__(self, api="https://api.openshift.com/api/",
             ca_path="/etc/pki/tls/certs/ca-bundle.crt", token=None):
 
         if token is None:
             raise Exception("Token was not supplied.")
 
+        self.thread_pool = ThreadPoolExecutor(4)
         self.base_api_uri = api
         self.auth_token = self.__confirm_auth(token)
 
         self.ca_certificate_path = ca_path
         if ca_path == False: # Used to Disable SSL warning (from each api call)
             requests.packages.urllib3.disable_warnings()
 
-        self.thread_pool = ThreadPoolExecutor(4)
-
 
     def __del__(self):
         self.thread_pool.shutdown(wait=True)
 
 
     def __confirm_auth(self, token):
-        r = requests.post('https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token', 
+        endpoint = 'https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token'
+        r = requests.post(endpoint,
                 data={'grant_type': 'refresh_token',
                     'client_id':'cloud-services',
                     'refresh_token': token})
 
         if r.status_code != 200:
-            raise Exception("""looking up infomation from: {}\n" 
-                    Error Code: {}""".format(endpoint, r.status_code))
+            if r.status_code == 400:
```

Comment:
This helps clear up a weird situation; and what is happening; when an offline token expires. 

In short they throw a 400 error (IDK why; to tell you that the offline token you have is bad/expired) and the body of the message tells you why.